includes/Abilities/Utilities/Date_Calculation.php

```diff
@@ -0,0 +1,493 @@
+						),
+						'timezone'    => array(
+							'type'        => 'string',
+							'description' => esc_html__( 'Timezone for calculations (e.g., "America/New_York"). Defaults to WordPress site timezone.', 'ai' ),
+						),
+					),
+					'required'   => array( 'pattern' ),
+				),
+				'output_schema'       => array(
+					'type'       => 'object',
+					'properties' => array(
+						'dates'   => array(
+							'type'        => 'array',
+							'description' => esc_html__( 'Calculated dates in ISO 8601 format.', 'ai' ),
+							'items'       => array(
+								'type'   => 'string',
+								'format' => 'date-time',
+							),
+						),
+						'pattern' => array(
+							'type'        => 'string',
+							'description' => esc_html__( 'The original pattern that was interpreted.', 'ai' ),
+						),
+					),
+				),
+				'execute_callback'    => array( $this, 'execute_calculate_dates' ),
+				'permission_callback' => array( $this, 'permission_callback' ),
+				'meta'                => array(
+					'show_in_rest' => true,
+					'mcp' => array(
+						'public' => true,
+						'type'   => 'tool',
+					),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Execute the calculate-dates ability.
+	 *
+	 * @since 0.2.0
+	 *
+	 * @param array<string, mixed> $input The input arguments.
+	 * @return array<string, mixed>|\WP_Error The calculated dates or error.
+	 */
+	public function execute_calculate_dates( array $input ) {
+		// Validate pattern is provided.
+		if ( empty( $input['pattern'] ) ) {
+			return new WP_Error(
+				'pattern_required',
+				esc_html__( 'A date pattern is required.', 'ai' )
+			);
+		}
+
+		// Sanitize and set defaults.
+		$pattern     = sanitize_text_field( $input['pattern'] );
+		$start_date  = isset( $input['start_date'] ) ? sanitize_text_field( $input['start_date'] ) : 'now';
+		$occurrences = isset( $input['occurrences'] ) ? absint( $input['occurrences'] ) : self::OCCURRENCES_DEFAULT;
+		$timezone    = isset( $input['timezone'] ) ? sanitize_text_field( $input['timezone'] ) : wp_timezone_string();
+
+		// Validate and limit occurrences.
+		$occurrences = min( max( $occurrences, 1 ), self::OCCURRENCES_MAX );
+
+		try {
+			// Calculate the dates.
+			$dates = $this->calculate_dates( $pattern, $start_date, $occurrences, $timezone );
+
+			return array(
+				'dates'   => $dates,
+				'pattern' => $pattern,
+			);
+		} catch ( Exception $e ) {
+			return new WP_Error(
+				'date_calculation_failed',
+				sprintf(
+				/* translators: %s: Error message. */
+					esc_html__( 'Failed to calculate dates: %s', 'ai' ),
+					$e->getMessage()
+				)
+			);
+		}
+	}
+
+	/**
+	 * Permission callback for date calculation.
+	 *
+	 * @since 0.2.0
+	 *
+	 * @param array<string, mixed> $args The input arguments.
+	 * @return bool|\WP_Error True if permitted, WP_Error otherwise.
+	 */
+	public function permission_callback( array $args ) {
+		// Anyone who can edit posts can use date calculations.
+		if ( ! current_user_can( 'edit_posts' ) ) {
```

Comment:
What is the thought process behind this permission check? It doesn't seem like we're interacting with any WordPress data so could argue no need for a capability check and maybe fine with just a user logged in check